src/Generators/MigrationGenerator.php

```diff
@@ -128,10 +128,10 @@ protected function populateStub(string $stub, Model $model): string
         return $stub;
     }
 
-    protected function populatePivotStub(string $stub, array $segments): string
+    protected function populatePivotStub(string $stub, array $segments, array $models = []): string
```

Comment:
Why is `models` needed?

Reply:
So the issue here is when two models have a Many to Many relationship, the pivot table was defaulting to `foreignId` which isn't correct if the Models has a `uuid` or `ulid` column type. This function checks the model `id` type and if its ulid it returns `foreignUlid`, `uuid` it returns `foreignUuid` and `foreignId` otherwise. 
E.g
Before 
```yaml
models:
  Diver:
    id: ulid primary
    name: string
    base_pay: decimal:10,2
    relationships:
      belongsToMany: Job
  Job:
    id: ulid primary
    location: string
    vessel: string
    relationships:
      belongsToMany: Diver
```
Would yield the pivot migration
```php
<?php

use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Support\Facades\Schema;

return new class extends Migration
{
    /**
     * Run the migrations.
     */
    public function up(): void
    {
        Schema::create('diver_job', function (Blueprint $table) {
            $table->foreignId('diver_id');
            $table->foreignId('job_id');
        });
    }

    /**
     * Reverse the migrations.
     */
    public function down(): void
    {
        Schema::dropIfExists('diver_job');
    }
};
```
After this change
```php
<?php

use Illuminate\Database\Migrations\Migration;
use Illuminate\Database\Schema\Blueprint;
use Illuminate\Support\Facades\Schema;

return new class extends Migration
{
    /**
     * Run the migrations.
     */
    public function up(): void
    {
        Schema::create('diver_job', function (Blueprint $table) {
            $table->foreignUlid('diver_id');
            $table->foreignUlid('job_id');
        });
    }

    /**
     * Reverse the migrations.
     */
    public function down(): void
    {
        Schema::dropIfExists('diver_job');
    }
};
```

Reply:
There is probably a cleaner way to do this. I will defer to you.

Reply:
I see. Thanks.